packages/editor/src/components/block-navigation/dropdown.js

```diff
@@ -31,10 +32,11 @@ function BlockNavigationDropdown() {
 					<IconButton
 						icon={ MenuIcon }
 						aria-expanded={ isOpen }
-						onClick={ onToggle }
+						onClick={ hasBlocks || ! isTextModeEnabled ? onToggle : undefined }
 						label={ __( 'Block Navigation' ) }
 						className="editor-block-navigation"
 						shortcut={ displayShortcut.access( 'o' ) }
+						aria-disabled={ ! hasBlocks || isTextModeEnabled }
```

Comment:
Can we use the disabled attribute instead of aria-disabled? I think it has the same effect for screen readers users and avoids the need for the onClick change done here as the onClick is not called if disabled is set to true.

Reply:
Thank you @jorgefilipecosta for tips.
Had adjusted details you've pointed.
And because this PR had been merged before:
https://github.com/WordPress/gutenberg/pull/12073/files
I did not touch any of those modifications regarding `onClick`. Only my changes had been discarded.

Reply:
@jorgefilipecosta - in my testing there is a subtle difference between `disabled` and `aria-disabled` - in case of using `disabled` you never have to tab through the disabled element which seems to be reasonable but differs from other buttons like `redo` or `undo`. I have already landed this PR, but given that we need to do something similar for Document Outline I can open a follow-up and include that, too. What do you think?